src/storage/standard_buffer_manager.cpp

```diff
@@ -522,6 +523,9 @@ void StandardBufferManager::DeleteTemporaryFile(block_id_t id) {
 	auto &fs = FileSystem::GetFileSystem(db);
 	auto path = GetTemporaryPath(id);
 	if (fs.FileExists(path)) {
+		auto handle = fs.OpenFile(path, FileFlags::FILE_FLAGS_READ);
+		auto content_size = handle->GetFileSize();
+		temporary_directory.handle->GetTempFile().DecreaseSizeOnDisk(content_size);
```

Comment:
Unlikely to really matter - but can we do this after removing the file instead of before?